activesupport/lib/active_support/core_ext/string/inflections.rb

```diff
@@ -164,8 +164,26 @@ def deconstantize
   #
   #   <%= link_to(@person.name, person_path) %>
   #   # => <a href="/person/1-donald-e-knuth">Donald E. Knuth</a>
-  def parameterize(sep = '-'.freeze)
-    ActiveSupport::Inflector.parameterize(self, sep)
+  #   
+  # To preserve the case of the characters in a string, use the `preserve_case` argument.
+  #
+  #   class Person
+  #     def to_param
+  #       "#{id}-#{name.parameterize(preserve_case: true)}"
+  #     end
+  #   end
+  #
+  #   @person = Person.find(1)
+  #   # => #<Person id: 1, name: "Donald E. Knuth">
+  #
+  #   <%= link_to(@person.name, person_path) %>
+  #   # => <a href="/person/1-Donald-E-Knuth">Donald E. Knuth</a>
+  def parameterize(sep = '', separator: '-', preserve_case: false)
+    separator = sep.presence || separator
+    if !sep.empty?
+      ActiveSupport::Deprecation.warn("Passing the separator argument as a positional parameter is deprecated and will soon be removed. Use `separator: '#{sep}'` instead.")
+    end
```

Comment:
```ruby
def parameterize(sep = nil, separator: '-', preserve_case: false)
  unless sep.nil?
    ActiveSupport::Deprecation.warn("Passing a separator as the first argument to `parameterize` is deprecated and will be removed in Rails 5.1. Use the `separator` keyword argument instead, e.g. `name.parameterize(separator: '_')`.")
    separator = sep
  end

  ...
```

We can do this because passing `nil` already doesn't work today (it raises), so if we see a non-nil first-argument, we can assume that the user is trying to pass a seperator.